packages/react/src/components/Forms/Select/mono.stories.tsx

```diff
@@ -134,6 +139,57 @@ export const SearchableUncontrolled = {
   },
 };
 
+export const AsyncSearchableUncontrolled = {
+  render: Template,
+
+  args: {
+    label: "Select a city",
+    options: async (context: ContextOptionAsaCallback) =>
+      new Promise((resolve) => {
```

Comment:
There is a hook problem I guess, the function keep being called, if you add a console.log, you can see it.

```suggestion
      new Promise((resolve) => {
        console.log("Keep calling");
```

A demo to show you:
[scrnli_3_19_2024_11-06-19 AM.webm](https://github.com/openfun/cunningham/assets/25994652/e2b5a658-ab78-471f-a3e5-31c07be6a718)

If I test it from the app perspective, my app bug totally.

Reply:
It's fixed. What I had to do inside mono-searchable.tsx was to update the code to move the callback execution to depend on the inputFilter state updates:

```
useEffect(() => {
      props.onSearchInputChange?.({ target: { value: inputFilter } });

      if (typeof props.options === "function") {
        (async () => {
          await computeOptionsToDisplay({ search: inputFilter });
        })();
      }
    }, [inputFilter]);
    ```